mountpoint-s3/src/inode/readdir.rs

```diff
@@ -65,33 +65,31 @@ impl ReaddirHandle {
         full_path: String,
         page_size: usize,
     ) -> Result<Self, InodeError> {
-        let local_entries = {
+        let cached_entries = {
             let inode = inner.get(dir_ino)?;
             let kind_data = &inode.get_inode_state()?.kind_data;
-            let local_files = match kind_data {
+            let cached_entries = match kind_data {
                 InodeKindData::File { .. } => return Err(InodeError::NotADirectory(dir_ino)),
-                InodeKindData::Directory { writing_children, .. } => writing_children.iter().map(|ino| {
-                    let inode = inner.get(*ino)?;
+                InodeKindData::Directory { children, .. } => children.values().map(|inode| {
```

Comment:
This seems problematic for large directories. Before we only needed to look at the children being written to, but now we need to look at every child before we can start streaming. So the first `ls` on a directory will be fast, but the second one will be slow.

Reply:
Okay, removed this part from this PR.